Add menu `auto` open and additional icon styles

B77Mills opened the pull request:
This ports the new style & flyout logic from the update version setup within pmmi's theme-monorail-leaders packages.  This port will allow us to use the current version of base-cms leaders while applying the new style & flyout direction logic.  It also adds support for flyouts on mobile.  

They card flyout will not account for available space and position on page to ensure it doesn't exceed the width of the window and that the card doesn't display off screen.

This will also allow us to update both default and monorail themes

Comparison(prod || defaultThemeDev || MonoRailDev
<img width="2970" alt="Screen Shot 2022-06-09 at 9 32 50 AM" src="https://user-images.githubusercontent.com/3845869/172872833-60096c6b-7bc9-42cd-89fe-d0aeb78e805f.png">

<img width="2524" alt="Screen Shot 2022-06-09 at 9 34 56 AM" src="https://user-images.githubusercontent.com/3845869/172873476-27ff61d9-d65d-421e-a29b-85f773b435e7.png">
<img width="2955" alt="Screen Shot 2022-06-09 at 9 36 06 AM" src="https://user-images.githubusercontent.com/3845869/172873516-9ce42e1d-1c39-4f00-8c4e-ca4111bade8e.png">

B77Mills committed:
port components from pmmi local pacakge

This will add the support for chevron icon set and the flyout auto support

B77Mills committed:
update card flyout logic to account for off page

When hoving over a company ensure the card in on screen.  If left link and the card positioned off screen move it left by the amount it is off screen.

B77Mills committed:
fix include

B77Mills committed:
On mobile only allow the card to be as wide as the window

ensure the card is no wider than the available window & add scroll to videos & products list

B77Mills committed:
update to new leaders style

zarathustra323 commented on packages/leaders-program/src/components/list/positions/menu-position.js:
```diff
@@ -15,9 +15,22 @@ class MenuPosition extends AbstractPosition {
-    if (this.opensLeft) return pageXOffset + calcs.link('left') - calcs.arrow('w') - calcs.menu('w');
-    return pageXOffset + calcs.link('right') + calcs.arrow('w');
+    if (this.opensAbove || this.opensBelow) {
+      // Determin if the card will display off screen and adjust posiiton accordinly
```
Clean up spelling in comments :)

zarathustra323 commented on packages/leaders-program/src/components/list/index.vue:
```diff
@@ -309,6 +318,23 @@ export default {
+    getAutoOpenDirection(content, linkRect) {
+      const avaiableRight = window.innerWidth - linkRect.right;
+      const shouldOpen = (avaiableRight >= window.innerWidth / 2) ? 'right' : 'left';
+      // Check there is enought space to open the content left/right or push below
```
Clean up spelling in comments :)

zarathustra323 commented on packages/leaders-program/src/components/containers/section.vue:
```diff
@@ -48,6 +58,8 @@
 <script>
 import PlusIcon from '../icons/add-circle-outline.vue';
 import MinusIcon from '../icons/remove-circle-outline.vue';
+import ChevronPlusIcon from '../icons/chevron-right.vue';
```
These should be named `ChevronRightIcon` and `ChevronDownIcon` (and also changed in the template to `<chevron-right-icon>` etc)

zarathustra323 commented on packages/leaders-program/src/components/containers/section.vue:
```diff
@@ -109,6 +123,10 @@ export default {
       type: String,
       default: 'Featured Products',
     },
+    iconSet: {
```
Change `iconSet` to `iconStyle` and explicitly set a default value to `plus-minus` and add as `v-else-if` check in the template

zarathustra323 replied:
Also, add a validator to the prop to only allow `plus-minus` and `chevron`

zarathustra323 commented on packages/leaders-program/src/scss/_variables.scss:
```diff
@@ -13,7 +13,7 @@ $leaders-gray-5: #f0f0f0 !default;
 // Base Styles
 $leaders-base-color: $leaders-black !default;
-$leaders-base-font-family: "IBM Plex Sans", sans-serif !default;
+$leaders-base-font-family: "Open Sans", sans-serif !default;
```
Are the fonts handled by the implementing site?

zarathustra323 commented on packages/leaders-program/src/scss/components/_card.scss:
```diff
@@ -6,6 +6,10 @@
   max-width: leaders-card-width();
   word-wrap: break-word;
   background-clip: border-box;
+  // On mobile set max width to veiwport
```
Cleanup spelling in comments

zarathustra323 commented on packages/leaders-program/src/scss/components/_section.scss:
```diff
@@ -50,15 +52,17 @@
         font-weight: $leaders-parent-section-content-title-font-weight;
         line-height: $leaders-parent-section-content-title-line-height;
-        color: $leaders-parent-section-content-title-color;
+        // color: $leaders-parent-section-content-title-color;
```
Remove commented out code

B77Mills committed:
Fix all my typos & remove commented out css.

B77Mills committed:
update iconStyle(iconSet) logic & prop validation

 - Change iconSet to iconStyle
 - Add validation on iconStyle with a default of plus-minus
 - Update v-id on icons.  Instead v-if v-else on wrapping element for display of button with icon.

zarathustra323 commented on packages/leaders-program/src/components/containers/section.vue:
```diff
@@ -109,6 +132,11 @@ export default {
+      type: String,
+      default: 'plus-minus',
+      validator: v => ['plus-minus', 'chevron'].includes(v),
+    },
```
Psuedo-code:

```vue
<template>
  <button
    class="leaders-section__toggle-button"
    @click="toggleExpanded"
  >
    <component :is="collapsedIcon" v-show="!isExpanded" :modifiers="iconModifiers" />
    <component :is="expandedIcon" v-show="isExpanded" :modifiers="iconModifiers" />
    <span class="leaders-section__toggle-button-title">{{ title }}</span>
  </button>
</template>

<script>
import PlusIcon from '../icons/add-circle-outline.vue';
import MinusIcon from '../icons/remove-circle-outline.vue';
import ChevronRightIcon from '../icons/chevron-right.vue';
import ChevronDownIcon from '../icons/chevron-down.vue';

export default {
  props: {
    iconStyle: {
      type: String,
      default: 'plus-minus',
      validator: v => ['plus-minus', 'chevron'].includes(v),
    },
  },
  computed: {
    expandedIcon() {
      if (this.iconStyle === 'chevron') return ChevronDownIcon;
      return MinusIcon;
    },
    collapsedIcon() {
      if (this.iconStyle === 'chevron') return ChevronRightIcon;
      return PlusIcon;
    },
  },
}
</script>
```

B77Mills committed:
Use computed component instead of v-if with button

Use a computed component to render the icon set instead of multipe v-if statments to handle which icon set to use.

zarathustra323 commented on packages/leaders-program/src/components/list/index.vue:
```diff
@@ -309,6 +318,23 @@ export default {
+      // Left:
+      // ContentWidth + 30px can not exceed link's left position
+      const canOpenLeft = shouldOpen === 'left' && (linkRect.left - content.getBoundingClientRect().width) >= 30;
+      if (canOpenRight || canOpenLeft) {
```
```js
if (canOpenRight || canOpenLeft) return shouldOpen;
return 'below';
```

zarathustra323 commented on packages/leaders-program/src/scss/components/_content-deck.scss:
```diff
@@ -16,6 +16,7 @@
   &__body {
     display: flex;
     flex-direction: row;
+    overflow-y: scroll;
```
`overflow-y: auto`

B77Mills committed:
update overflow-y: scroll to auto

B77Mills committed:
update GetAutoOpenDirection to return value

Update getter to return value and set this.openDirection = this.getAutoOpenDirection();
